src/lib/PwUp.ts

```diff
@@ -0,0 +1,196 @@
+import { PwUpConfig, PwUpTypes, SudtCell } from "./PwUpTypes";
+import { BI, Cell, config, core, helpers, Indexer, RPC, toolkit, utils, Address } from "@ckb-lumos/lumos";
+import { default as createKeccak } from "keccak";
+
+export const CONFIG = config.createConfig({
+  PREFIX: "ckt",
+  SCRIPTS: {
+    ...config.predefined.AGGRON4.SCRIPTS,
+    // https://github.com/lay2dev/pw-core/blob/861310b3dd8638f668db1a08d4c627db4c34d815/src/constants.ts#L156-L169
+    PW_LOCK: {
+      CODE_HASH: "0x58c5f491aba6d61678b7cf7edf4910b1f5e00ec0cde2f42e0abb4fd9aff25a63",
+      HASH_TYPE: "type",
+      TX_HASH: "0x57a62003daeab9d54aa29b944fc3b451213a5ebdf2e232216a3cfed0dde61b38",
+      INDEX: "0x0",
+      DEP_TYPE: "code",
+    },
+  },
+});
+
+config.initializeConfig(CONFIG);
+
+const CKB_RPC_URL = "https://testnet.ckb.dev/rpc";
+const CKB_INDEXER_URL = "https://testnet.ckb.dev/indexer";
+const rpc = new RPC(CKB_RPC_URL);
+const indexer = new Indexer(CKB_INDEXER_URL, CKB_RPC_URL);
+
+// prettier-ignore
+interface EthereumRpc {
+    (payload: { method: 'personal_sign'; params: [string /*from*/, string /*message*/] }): Promise<string>;
+}
+
+// prettier-ignore
+export interface EthereumProvider {
+    selectedAddress: string;
+    isMetaMask?: boolean;
+    enable: () => Promise<string[]>;
+    addListener: (event: 'accountsChanged', listener: (addresses: string[]) => void) => void;
+    removeEventListener: (event: 'accountsChanged', listener: (addresses: string[]) => void) => void;
+    request: EthereumRpc;
+}
+
+// @ts-ignore
+export const ethereum = window.ethereum as EthereumProvider;
+
+export function asyncSleep(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export class PwUp implements PwUpTypes {
+  config: PwUpConfig;
+  isConnected: boolean;
+  constructor(config: PwUpConfig) {
```

Comment:
We can use a simpler constructor like this

```ts
  constructor(network: NetworkType) {
   ...  
  }
```

Reply:
good advise, fixed